aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -481,6 +623,121 @@ module aptos_framework::delegation_pool {
 
         // save delegation pool ownership and resource account address (inner stake pool address) on `owner`
         move_to(owner, DelegationPoolOwnership { pool_address });
+
+        // All delegation pool enable partial governace voting by default once the feature flag is enabled.
+        if (features::partial_governance_voting_enabled() && features::delegation_pool_partial_governance_voting_enabled()) {
+            enable_partial_governance_voting(pool_address);
+        }
+    }
+
+    /// Enable partial governance voting on a stake pool. The voter of this stake pool will be managed by this module.
+    /// THe existing voter will be replaced. The function is permissionless.
+    public entry fun enable_partial_governance_voting(
+        pool_address: address,
+    ) acquires DelegationPool, GovernanceRecords {
+        assert!(features::partial_governance_voting_enabled(), error::invalid_state(EDISABLED_FUNCTION));
+        assert!(features::delegation_pool_partial_governance_voting_enabled(), error::invalid_state(EDISABLED_FUNCTION));
+        assert_delegation_pool_exists(pool_address);
+        // synchronize delegation and stake pools before any user operation.
+        synchronize_delegation_pool(pool_address);
+
+        let delegation_pool = borrow_global<DelegationPool>(pool_address);
+        let stake_pool_signer = retrieve_stake_pool_owner(delegation_pool);
+        // delegated_voter is managed by the stake pool itself, which signer capability is managed by DelegationPool.
+        // So voting power of this stake pool can only be used through this module.
+        stake::set_delegated_voter(&stake_pool_signer, signer::address_of(&stake_pool_signer));
```

Comment:
I think it should add that the voter address has to be itself because aptos_governance needs it.

Reply:
Sry I don't fully understand it. Do you mean adding it into comments? Can you elaborate "aptos_governance needs it"?